packages/@aws-cdk/aws-servicecatalogappregistry/lib/application.ts

```diff
@@ -42,6 +56,45 @@ export interface ApplicationProps {
 abstract class ApplicationBase extends cdk.Resource implements IApplication {
   public abstract readonly applicationArn: string;
   public abstract readonly applicationId: string;
+  private readonly associatedAttributeGroups: Set<string> = new Set();
+  private readonly associatedResources: Set<string> = new Set();
+
+  /**
+   * Associate an attribute group with application
+   * If the attribute group is already associated, it will ignore duplicate request.
+   */
+  public associateAttributeGroup(attributeGroup: IAttributeGroup): void {
+    if (!this.associatedAttributeGroups.has(attributeGroup.node.addr)) {
+      const hashId = this.generateUniqueHash(attributeGroup.node.addr);
+      new CfnAttributeGroupAssociation(this, `AttributeGroupAssociation${hashId}`, {
+        application: this.applicationId,
+        attributeGroup: attributeGroup.attributeGroupId,
+      });
+      this.associatedAttributeGroups.add(attributeGroup.node.addr);
+    }
+  }
+
+  /**
+   * Associate a stack with the application
+   * If the resource is already associated, it will ignore duplicate request.
+   * A stack can only be associated with one application.
+   */
+  public associateStack(stack: cdk.Stack): void {
+    if (!this.associatedResources.has(stack.node.addr)) {
+      const hashId = this.generateUniqueHash(stack.node.addr);
+      new CfnResourceAssociation(this, `ResourceAssociation${hashId}`, {
+        application: this.applicationId,
+        resource: stack.stackId,
+        resourceType: 'CFN_STACK',
+      });
+      this.associatedResources.add(stack.node.addr);
+    }
+  }
+
+  /**
+   * Create a unique id
+   */
+  protected abstract generateUniqueHash(value: string): string;
```

Comment:
I don't love the name of the `value` parameter - too generic.

How about `resourceAddress`?

Reply:
sure, copying over from `servicecatalog` where we would have different values passed in but here it is more simple.